Jropp/jason-ropp-js-iobp
# Onboarding Project: App-Cleanup

## What It Does

The app is now consolidated into a single page. It opens to the list of users. From there you can edit existing users or create new users.

The former design was less intuitive and required clicking a link and static page loads to switch between creating a new user and seeing the list of users.

The users-list component became the parent component. I used dom-if templates and a modes property to set the state of the app.  Clicking on 'create new user', 'save', 'cancel', etc alters the state of the app rather than making page changes.

Closes #47 

## How To Test

1. Create a new user and save. It should display immediately in the user-list
2. Edit a user's data and save. It should display immediately.
3. Delete all users. You should be prompted that the database is empty and to create a new user. 
4. Create a new user but cancel instead of saving. You should see the user display page again.
5. When there are no users in the database, start filling out the new user fields. Click cancel. The fields should be emptied out. 


## Notes/Caveats

This still implements local storage for the database. This will be changed in #48 

## Checklist

Under penalty of public shaming, I avow that I:


Browsers tested:

## Dependencies

none
